changelog.d/8093.bugfix

```diff
@@ -0,0 +1 @@
+Return the previous stream token if a non-member event is a duplicate.
```

Comment:
This is an awful changelog entry for a bugfix, but I don't know what an admin / user might see in this situation. It seems to only be an issue in replication, but I'm not 100% sure.

Reply:
I wonder if this is an actual issue, as if this is only a problem in create room then we won't ever hit the "duplicate state" path. So maybe its not a bug fix and just a misc fix to type hinting?

Reply:
Oops, I missed this sorry. I think you're correct that it is probably just a misc, not a "real" bug.

Reply:
I updated this in 408aef827691be3a1b92f38ebdf5aab7468ab5ef.